uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/form/components/form-sections/leave-type-wizard-form-settings.html

```diff
@@ -0,0 +1,52 @@
+<div class="row">
+  <div class="col-md-3">
+    <ul class="nav nav-pills nav-stacked">
+      <li ng-repeat="settingsTab in form.settingsTabs"
+        ng-class="{ active: settingsTab.active }"
+        ng-click="form.openSettingsTab(settingsTab.name)">
+        <a href="#">
+          {{settingsTab.label}}
+        </a>
+      </li>
+    </ul>
+  </div>
+  <div class="col-md-9">
+    <div class="row" ng-repeat="field in form.getSettingsTabFields()">
+      <div class="col-md-3">
+        <label>
+          {{field.label}}
+          <help-text ng-if="field.helpText">
+            <p class="respect-line-breaks">{{field.helpText}}</p>
+          </help-text>
+        </label>
+      </div>
+      <div class="col-md-9">
+        [template of the field]
+      </div>
+    </div>
+    <hr/>
+    <div class="row">
+      <div class="col-md-12">
+        <div class="btn btn-secondary pull-right"
+          ng-if="!form.isOnLastSettingsTab"
+          ng-click="form.openSettingsTab(form.settingsTabs[form.activeSettingsTabIndex + 1].name)">
```

Comment:
why the need to delegate the indexing logic to the view? Something like this
```html
ng-click="form.openNextTab()"
```
would be much cleaner and you would keep the pagination logic in the component controller instead of leaking it into the view

Reply:
but you called it `openNextSettingsTab`, thus tightly coupling to a specific section. If there isn't a technical reason for why your component needs to rely on hardcoded references, then you need to remove those references

Reply:
Not anymore.